packages/core/database/src/migrations/index.ts

```diff
@@ -7,9 +7,12 @@ import type { Database } from '..';
 export type { MigrationProvider };
 
 export const createMigrationsProvider = (db: Database): MigrationProvider => {
-  const providers = [createUserMigrationProvider(db), createInternalMigrationProvider(db)];
+  const userProvider = createUserMigrationProvider(db);
+  const internalProvider = createInternalMigrationProvider(db);
+  const providers = [userProvider, internalProvider];
 
   return {
+    internal: internalProvider,
```

Comment:
worth transforming this into a map so you could do strapi.db.migrations.providers.internal.register

(we could allow user programmaticaly registering later this way)

Reply:
an also we could make the providers a map so you could add a provider & we just loop over providers.values() ...

Reply:
Is there a good way to strictly type the keys of a map? I have something like `Map<string, MigrationProvider>;` atm, but that leaves the typings a bit loosen .

`strapi.db.migrations.providers.`  <-- does not autocomplete to internal

Reply:
well, going to propose something that makes sense to me! let me know what you think (in a couple of minutes :P )